pkg/clusteragent/autoscaling/workload/model/pod_autoscaler.go

```diff
@@ -148,16 +159,36 @@ func (p *PodAutoscalerInternal) UpdateFromSettings(podAutoscalerSpec *datadoghq.
 	p.horizontalEventsRetention = getHorizontalEventsRetention(podAutoscalerSpec.Policy, longestScalingRulePeriodAllowed)
 }
 
-// UpdateFromValues updates the PodAutoscalerInternal from a new scaling values
+// UpdateFromValues updates the PodAutoscalerInternal scaling values
```

Comment:
Ideally we would need to refine this API so that it's not complete freedom from controller 🤔

Reply:
Could this validation (ensuring that `scalingValues` is always either `mainScalingValues` or `fallbackScalingValues`) be done in one of the follow-up PRs to make use of `fallbackScalingValues`/`mainScalingValues`? There are no changes to the controller logic yet so `scalingValues` is the only field being used at the moment

Reply:
That works yes!